org.eclipse.transformer/src/main/java/org/eclipse/transformer/action/impl/SignatureRuleImpl.java

```diff
@@ -181,6 +197,37 @@ public BundleData getBundleUpdate(String symbolicName) {
 	public String getDirectString(String initialValue) {
 		return directStrings.get(initialValue);
 	}
+        
+        private final Map<String, Map<String, String>> perClassConstantStrings;
+
+        @Override
+        public String getConstantString(String initialValue, String clazz) {
+            Map<String, String> m = perClassConstantStrings.get(clazz);
+            if (m != null) {
+                // Do we have a direct mapping?
+                String full = m.get(initialValue);
+                if (full != null) {
+                    System.out.println("DIRECT REPLACEMENT:[" + clazz + "]" + initialValue + "=>" + full);
+
+                    return full;
+                }
+                String transformedString = initialValue;
+                boolean transformed = false;
+                // Transform all tokens possibly present in the inputValue
+                for (String k : m.keySet()) {
+                    if (transformedString.contains(k)) {
```

Comment:
I'd add a comment here to indicate that the 'contains' test might be changed to use regular expressions in the future.

Reply:
Thanks, done.